crates/hir/src/display.rs

```diff
@@ -178,7 +178,7 @@ impl HirDisplay for Function {
                 AssocItemContainer::Impl(_) => "impl",
             };
             write!(f, "\n    // Bounds from {container_name}:",)?;
-            write_where_predicates(&container_params, &data.store, f)?;
```

Comment:
the type confusion here seems... suboptimal. can do this later, but would be nice to make this impossible to express in the future.

Reply:
Unfortunately I don't see how. It's not even a confusion between a body and a signature store, it's between a signature store and the signature store of the container.